src/html-media-source.js

```diff
@@ -187,6 +199,7 @@ export default class HtmlMediaSource extends videojs.EventTarget {
       }
 
       this.player_.on('mediachange', this.onPlayerMediachange_);
+      this.player_.on('seeking', this.onPlayerSeeking_);
```

Comment:
I'm not sure if it matters, but the captions should all be reset before it starts ingesting the video. Don't know if the `seeked` event will fire and complete before data is `push`ed through the transmuxer.